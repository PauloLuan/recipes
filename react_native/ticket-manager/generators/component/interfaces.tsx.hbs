import { I{{capitalize componentName}}Props } from 'native-base'

export interface {{capitalize componentName}}Props extends I{{capitalize componentName}}Props {
  testId?: string
}
