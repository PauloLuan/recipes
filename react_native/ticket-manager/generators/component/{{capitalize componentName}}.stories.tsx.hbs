import { storiesOf } from '@storybook/react-native'
import React from 'react'
import { {{capitalize componentName}} } from './{{capitalize componentName}}'

storiesOf('Components/{{capitalize componentName}}', module).add('default: ', () => <{{capitalize componentName}} />)
