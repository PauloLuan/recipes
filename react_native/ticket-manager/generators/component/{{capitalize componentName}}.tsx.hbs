import React from 'react'
import { I{{capitalize componentName}}Props, {{capitalize componentName}} as NativeBase{{capitalize componentName}} } from 'native-base'

export interface {{capitalize componentName}}Props extends I{{capitalize componentName}}Props {
  testId?: string
}

export const {{capitalize componentName}} = ({
  testId = '{{kebabCase componentName}}-id',
  ...props
}: {{capitalize componentName}}Props) => {
  return (
    <NativeBase{{capitalize componentName}} testID={testId} {...props}></NativeBase{{capitalize componentName}}>
  )
}
