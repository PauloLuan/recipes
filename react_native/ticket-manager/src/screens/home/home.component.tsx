import React from 'react'
import * as S from './home.styles'

export interface homeProps {
  testId?: string
}

export const Home = ({ testId = 'component-home-id', ...props }: homeProps) => {
  return (
    <S.Wrapper testID={testId} {...props}>
      <S.Heading>teste</S.Heading>
    </S.Wrapper>
  )
}
