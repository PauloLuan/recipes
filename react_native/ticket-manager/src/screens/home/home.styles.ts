import styled from 'styled-components/native'

export const Wrapper = styled.View`
  flex: 1;
  background-color: ${({ theme }) => theme.colors.background};
`

export const Heading = styled.Text`
  font-size: 24px;
  font-family: ${({ theme }) => theme.fonts.primaryBold};
`
