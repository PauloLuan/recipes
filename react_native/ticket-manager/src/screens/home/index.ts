export * from './home.component'
export * as HomeStyles from './home.styles'
