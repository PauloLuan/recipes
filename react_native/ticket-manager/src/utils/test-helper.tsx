import { render, RenderAPI } from '@testing-library/react-native'
import { NativeBaseProvider } from 'native-base'
import React from 'react'
import { theme } from '../global/styles/theme'

export const renderWithTheme = (children: React.ReactNode): RenderAPI =>
  render(<NativeBaseProvider theme={theme}>{children}</NativeBaseProvider>)
