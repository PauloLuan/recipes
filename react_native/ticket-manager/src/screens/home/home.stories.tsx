import { storiesOf } from '@storybook/react-native'
import React from 'react'
import { Home } from './home.component'

storiesOf('Components/Home', module).add('default: ', () => <Home />)
