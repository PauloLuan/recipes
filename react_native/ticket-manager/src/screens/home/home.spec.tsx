import { waitFor } from '@testing-library/react-native'
import React from 'react'
import { renderWithTheme } from '../../utils/test-helper'
import { Home } from './home.component'

describe('<home />', () => {
  const testId = 'component-home-id'

  it('should render the home component', async () => {
    const { queryByTestId, toJSON } = renderWithTheme(<Home testId={testId} />)
    await waitFor(() => expect(queryByTestId(testId)).toBeTruthy())
    expect(toJSON()).toMatchSnapshot()
  })
})
