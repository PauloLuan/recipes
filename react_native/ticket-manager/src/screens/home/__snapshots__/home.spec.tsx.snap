// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<home /> should render the home component 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#f8f9fa",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
  testID="component-home-id"
>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "sens-bold",
          "fontSize": 24,
        },
      ]
    }
  >
    teste
  </Text>
</View>
`;
